### PR TITLE
Add stacked-bar chart type

### LINKED ISSUE #52: Add stacked bar chart type
# FEATURE REQUEST / CODE CHANGE

## Summary
Add a chart type that can handle ordinal x-axis data with multiple series and render it as a stacked bar chart.

## Motivation
UX request to be able to display categories of data as color-coded stacked bars.

## Detailed design
In order to render stacked bars from ordinal data, new d3 helpers will need to be created to make use of d3's ordinal / band scale features. The d3 stack layout will be necessary as well to transform the input data's multiple series into the stack format d3 prefers to use.

New properties on the component will be required that define which fields of the input data should be mapped to data series, and to allow the user to define series colors.

The assumption is being made that input data will be an array of objects each containing the same set of keys; one for the x-axis labels and one for each series of data having numeric values.

## How we teach this
An example will be included in the dummy app.

### PR DESCRIPTION
**Pull request by twjohnston**
# Overview

## Summary
Add a new stacked-bar chart type and the d3 ordinal/scale/stacked data features required to support it.

## Issue Number(s)
* Closes #52 

## Screenshots or recordings
dummy

## Checklist
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] I have evaluated if the _README.md_ documentation needs to be updated
* [x] I have evaluated if the _/tests/dummy/_ app needs to be modified
* [ ] I have evaluated if DocBlock headers needed to be added or updated
* [x] I have verified that lint and tests pass locally with my changes
* [ ] If a fork of a dependent package had to be made to address the issue this PR closes:
  * [ ] I noted in the fork's _README.md_ the reason the fork was created
  * [ ] I have opened an upstream issue detailing what was deficient about the dependency
  * [ ] I have opened an upstream PR addressing this deficiency
  * [ ] I have opened an issue in this repository to track this PR and schedule the removal of the usage of the fork

# Semver
**This project uses [semver](http://semver.org), please check the scope of this PR:**

- [ ] #none#
- [ ] #patch#
- [x] #minor#
- [ ] #major#

# CHANGELOG
- Add stacked-bar chart type

### CODE REVIEW

File: addon/components/frost-chart-svg-plot-stacked-bar.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import {stack as d3Stack} from 'd3-shape'
+import computed, {readOnly} from 'ember-computed-decorators'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
+
+import layout from '../templates/components/frost-chart-svg-plot-stacked-bar'
+
+const defaultSeriesColors = [
```
Comment (notmessenger) [inline addon/components/frost-chart-svg-plot-stacked-bar.js]: Did you intend to use this elsewhere in the code, or perhaps once was, but not longer am?  Line 43 assigns this data to another property name so it seems that this array should just be defined there as well without being assigned to a separate property name that is not re-used.

File: addon/templates/components/frost-chart-svg-plot-stacked-bar.hbs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{{#each _bars key='@index' as |bar|}}
```
Comment (notmessenger) [inline addon/templates/components/frost-chart-svg-plot-stacked-bar.hbs]: What purpose does `key='@index'` serve?

File: app/helpers/d3-band-scale.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Re-export the d3-band-scale helper into the app namespace
+ */
+export {default, d3BandScale} from 'ember-frost-chart/helpers/d3-band-scale'
```
Comment (notmessenger) [inline app/helpers/d3-band-scale.js]: Should just be able to do `default` and not also `d3BandScale`, right?

File: app/helpers/ordinal-ticks.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Re-export the ordinal-ticks helper into the app namespace
+ */
+export {default, ordinalTicks} from 'ember-frost-chart/helpers/ordinal-ticks'
```
Comment (notmessenger) [inline app/helpers/ordinal-ticks.js]: Should just be able to do `default` and not also `ordinalTicks`, right?

File: tests/integration/components/frost-chart-svg-plot-stacked-bar-test.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+  beforeEach(function () {
+    this.setProperties({
+      data,
+      xDomain,
+      yDomain
+    })
+
+    this.render(hbs`
+      <style>
```
Comment (notmessenger) [inline tests/integration/components/frost-chart-svg-plot-stacked-bar-test.js:27]: Is this styling necessary for the test?

**Review by notmessenger — CHANGES_REQUESTED** [review]

File: addon/templates/components/frost-chart-svg-plot-stacked-bar.hbs (hunk 2 above)
Comment (twjohnston) [reply addon/templates/components/frost-chart-svg-plot-stacked-bar.hbs]: Good question... that was a copy/paste from existing code, but I don't think it's actually doing anything. I'll remove it.

File: tests/integration/components/frost-chart-svg-plot-stacked-bar-test.js (hunk 5 above)
Comment (twjohnston) [reply tests/integration/components/frost-chart-svg-plot-stacked-bar-test.js:27]: Without it the SVG element won't have a proper height and will throw errors in the console, though the tests will still pass. Including the style removes the console errors and renders a real looking chart for the integration tests.

Comment (notmessenger) [reply tests/integration/components/frost-chart-svg-plot-stacked-bar-test.js:27]: Do we need to point this styling requirement out in the dummy app so users know to apply (appropriate) height to their chart?

Comment (twjohnston) [reply tests/integration/components/frost-chart-svg-plot-stacked-bar-test.js:27]: Good idea, I added a note to the dummy app.

**Review by notmessenger — DISMISSED** [review]

**Review by juwara0 — DISMISSED** [review]

**Review by juwara0 — APPROVED** [review]

**Pull request merged**